### PR TITLE
ReferableUtils

### LINKED ISSUE #155: [TICKET] Allow idShorts with regexp `/__\d{1,}__$/`
> As a BaSyx developer
I would like to have support for idShorts with regexp `/__\d{1,}__$/`

## Rules
- Adaption/Extension of `checkIdShort()`
- Creation of tests/test examples

## Entry Points
- https://github.com/eclipse-basyx/basyx-aas-web-ui/blob/main/aas-web-ui/src/mixins/SubmodelElementHandling.ts#L141
- https://github.com/eclipse-basyx/basyx-aas-web-ui/blob/main/aas-web-ui/tests/mixins/SubmodelElementHandling.test.ts

## Acceptance Criteria
- Tests have been extended
- Tests are being successfull

### PR DESCRIPTION
**Pull request by seicke**
- Moved ~~checkSemanticId()~~ `checkIdShort()` from IDUtils (cause methods scope is referables)
- Add tests for all ReferableUtils methods
- Fixes of ReferableUtils methods regarding tests

Closes #155

### CODE REVIEW

File: aas-web-ui/src/utils/ReferableUtils.ts
Comment (aaronzi) [inline aas-web-ui/src/utils/ReferableUtils.ts:1]: For possible future support of multiple languages, could you please add a parameter to specify the language? The parameter should have a default set to 'en'

File: aas-web-ui/tests/utils/Referable.test.ts [hunk 1]
```diff
@@ -0,0 +1,957 @@
...
+            startsWith: true,
+            strict: true,
+            result: false,
+        },
+    ];
+
+    // Tests for checkSemanticId()
+    checkSemanticIdTestData.forEach(function (checkSemanticIdTestDataset) {
+        // Define test data
+        const referable = checkSemanticIdTestDataset.referable; //e.g. the ID of a ConceptDescription
```
Comment (aaronzi) [inline aas-web-ui/tests/utils/Referable.test.ts:946]:
```suggestion
        const referable = checkSemanticIdTestDataset.referable; // e.g. the ID of a ConceptDescription
```

**Review by aaronzi — CHANGES_REQUESTED** [review]

**Comment by aaronzi** [comment]
Thank you for the PR :)

> * Moved checkSemanticId from IDUtils (cause methods scope is preferable)

you mean checkIdShort, right?
Please correct the names of all the variables where this naming mistake was made.

**Comment by seicke** [comment]
Sorry for mixing-up the method names!
Yes, I mean checkIdShort.

**Pull request merged**